ghost-cat-app/src/app/Proxy/ServerFacade.ts

```diff
@@ -19,6 +21,87 @@ export class ServerFacade {
     return this.http.post<ImageQueryResponse>(this.url, imageQueryRequest);
   }
 
+  public async login(
+    userIDInput: string | null,
+    passwordHashInput: string | null,
+    router:Router
+  ): Promise<void> {
+    let url =
+      'https://az5x52mixa.execute-api.us-east-1.amazonaws.com/dev/login';
+
+    let loginRequest: JSON;
+    let resp: any;
+
+    //working test user
+    /*obj: any = 
+    {
+      "userID": "researcherID",
+      "passwordHash":"r4greog5httr"
+    };*/
+
+
+    let obj: any = 
+    {
+      "userID": userIDInput,
+      "passwordHash":passwordHashInput
+    };
+
+    loginRequest = <JSON>obj;
+
+      this.http
+      .post<string>(url, loginRequest)
+      .pipe(catchError(this.handleError('login')))
+      .subscribe({
+        next(response) {
+          resp = JSON.parse(JSON.stringify(response));
+          if(resp.auth_token != null){
+          
+            //alert('Got success: ' + JSON.stringify(response));
+            router.navigate([`/search`]);
+          }
+          else{
+            alert('Error while logging in: ' + JSON.stringify(response));
+          }
+        },
+      });
+  }
+
+  public async register(
+    userIDInput: string | null,
```

Comment:
Same comment here as above. I think it would be best to make a `RegisterRequest` class that could contain all of these parameters and member variables.